docs/api-spec.md

```diff
@@ -0,0 +1,322 @@
+| `startedAt`* | string     | REQUIRED    |
+
+\* `startedAt` MUST be populated by the implementation, in [RFC3339](https://tools.ietf.org/html/rfc3339).
+
+### `ContainerStateWaiting`
+
+| Field Name | Field Type | Requirement |
+|------------|------------|-------------|
+| `reason`   | string     | REQUIRED    |
+| `message`  | string     | REQUIRED    |
+
+### `ContainerStateTerminated`
+
+| Field Name    | Field Type | Requirement |
+|---------------|------------|-------------|
+| `exitCode`    | int32      | REQUIRED    |
+| `reason`      | string     | REQUIRED    |
+| `message`     | string     | REQUIRED    |
+| `startedAt`*  | string     | REQUIRED    |
+| `finishedAt`* | string     | REQUIRED    |
+
+\* `startedAt` and `finishedAt` MUST be populated by the implementation, in [RFC3339](https://tools.ietf.org/html/rfc3339).
+
+### `EnvVar`
+
+| Field Name | Field Type | Requirement |
+|------------|------------|-------------|
+| `name`     | string     | REQUIRED    |
+| `value`    | string     | REQUIRED    |
+
+**NB:** All other [EnvVar](https://godoc.org/k8s.io/api/core/v1#EnvVar) types inherited from [core.v1/EnvVar](https://godoc.org/k8s.io/api/core/v1#EnvVar) and supported by the Kubernetes implementation (e.g., `valueFrom`) are **OPTIONAL** for the purposes of this spec.
+
+### `Param`
+
+| Field Name | Field Type      | Requirement |
+|------------|-----------------|-------------|
+| `name`     | string          | REQUIRED    |
+| `value`    | `ArrayOrString` | REQUIRED    |
+
+### `ParamSpec`
+
+| Field Name    | Field Type | Requirement |
+|---------------|------------|-------------|
+| `name`        | string     | REQUIRED    |
+| `description` | string     | REQUIRED    |
+| `type`        | Enum: <br>- `"string"` (default) <br>- `"array"` | REQUIRED |
+| `default`     | `ArrayOrString` | REQUIRED |
+
+### `Step`
+
+| Field Name   | Field Type | Requirement |
+|--------------|------------|-------------|
+| `name`       | string     | REQUIRED    |
+| `image`      | string     | REQUIRED    |
+| `args`       | []string   | REQUIRED    |
+| `command`    | []string   | REQUIRED    |
+| `workingDir` | []string   | REQUIRED    |
+| `env`        | `[]EnvVar` | REQUIRED    |
+| `script`     | string     | REQUIRED    |
+
+**NB:** All other fields inherited from the [core.v1/Container](https://godoc.org/k8s.io/api/core/v1#Container) type supported by the Kubernetes implementation are **OPTIONAL** for the purposes of this spec.
+
+### `StepState`
+
+| Field Name    | Field Type                 | Requirement |
+|---------------|----------------------------|-------------|
+| `name`        | string                     | REQUIRED    |
+| `imageID`     | string                     | REQUIRED    |
+| `waiting`*    | `ContainerStateWaiting`    | REQUIRED    |
+| `running`*    | `ContainerStateRunning`    | REQUIRED    |
+| `terminated`* | `ContainerStateTerminated` | REQUIRED    |
+| `taskResults` | `TaskRunResult`            | REQUIRED    |
```

Comment:
[]TaskRunResult?

Reply:
( dont let this fool you into thinking i was this thorough with everything i looked at XD )

Reply:
Had me fooled! 😄